# Harsh_Gotecha_ BKBI_MBA_15

# Assignment for numeric 

# Write a Python function to find the maximum of three numbers from the given list

In [3]:
h = 1,2,3,2,1
max(h)

3

# 2. Write a Python function to sum all the numbers in a list.
Sample List : (8, 2, 3, 0, 7)
    

In [3]:
a = (8, 2, 3, 0, 7)
sum(a)

20

# 3. Write a Python function to multiply all the numbers in a list.
Sample List : (8, 2, 3, -1, 7)

In [5]:
8*2*3*-1*7

-336

# 5. Evaluate following expressions:

In [8]:
7 - 24 / 8 * 4 + 6 

1.0

In [9]:
18 / 3 - 7 + 2 * 5

9.0

In [11]:
6 * 4 / 12 + 72 / 8 - 9

2.0

In [13]:
(17 - 6 / 2) + 4 * 3

26.0

In [19]:
(3 * 5 **2 / 15) - (5 - 2 ** 2 )

4.0

In [21]:
(22 / 2 - 2.5)** 2 + ( 4 - 6 / 6) ** 2

81.25

# Variables Assignment

# 1. Write a python program to calculate the area of a rectangle given its length and width.

In [13]:
length = int(input("Enter the length of rectengle"))
width = int(input("Enter the width of rectengle")) 

Enter the length of rectengle12
Enter the width of rectengle25


In [16]:
print("Area of a rectangle is",length*width)


Area of a rectangle is 300


# 2. Create a Program that takes a user's name and birth year as input and prints a greeting message.
I/P: Enter User Name : Tushar
Enter Year of Birth : 1990
O/P: Hi Tushar, Today you are 23 years Old, Welcome!

In [1]:
name= input("Enter user name")
birth= input("Enter year of birth")


print(f'Hi {name}, Today you are {birth} years old, welcome!')


Enter user nameHarsh
Enter year of birth21
Hi Harsh, Today you are 21 years old, welcome!


# 3. Calculate the compound interest for a given principal amount, interest rate and time period

In [1]:
principal = int(input("Enter the principal"))
rate = int(input("Enter the rate"))
years = int(input("Enter the years"))

Compound_int= principal*(1+(rate/100))**years-principal
Compound_int



Enter the principal20000
Enter the rate12
Enter the years2


5088.000000000004

# 4. Implement a program that swaps the values of two variables with the help of third variable.

In [1]:
a = 1 + 2
b = a - 2
a = a - b

a,b

(2, 1)

# 5. Write a Python Program to calculate cube of the number

In [1]:
h= int(input("Enter the value"))

print(h**3)
pow(h,3)

Enter the value5
125


125

# 6. Create variables for storing a Person's name, age and average test score.

In [1]:
personalname = input("Enter you name")
age = input("Enter your age")
avgscore= input("Enter your average test score")

print(f'Hi {personalname}, your age is {age} and your average test score is {avgscore}')


Enter you nameharsh
Enter your age20
Enter your average test score50
Hi harsh, your age is 20 and your average test score is 50


# 7. Create a Program that takes a temperature in Celcius and converts it to Kelvin.
 

In [2]:
celcius = int(input("celcius is "))

print("Kelvin is ",celcius * 273.15)


celcius is 5
Kelvin is  1365.75


# 8. Calculate the area and circumference of a circle given its radius.
 

In [1]:
r= int(input("Enter the R value"))

area = 3.14159265359*r**2


circumference = 2*3.14159265359*r
area, circumference 


Enter the R value1231


(4760646.985136796, 7734.60111313858)

# 9. Implement a Program that converts a given number of minutes into hours and minutes.
Sample I/P : 130 Minutes
O/P : 1 Hour 10 Minutes

In [12]:
minutes = int(input("Enter the minutes"))

hour = int(minutes/60)

minute = minutes%60

print(hour,'hour',minute,'minute')

Enter the minutes130
2 hour 10 minute


# 10. Write a Program that converts a given number of days into years, weeks, and days.
Sample I/P : 375 Days
O/P : 1 Years, 1 Week, 3 Days

In [5]:
day = int(input("Enter number of day"))

year = day//365
week = day%365//7
day = day%365%7

print(year,'Years',week,'week',day,'days')

Enter number of day375
1 Years 1 week 3 days


# 11. Implement a program that swaps the values of two variables without using 3rd variable.

# 12. Write a Program to check the data type of below variables:

In [13]:
a = 1.90
b = 5/45
c = "DSA"
d = 'Amir'
e = 'True'
f = True
g = 7 + 2j
h = 3712332
i = '2023-08-04'

type(a), type(b), type(c),type(d),type(e),type(f),type(g),type(h),type(i)

(float, float, str, str, str, bool, complex, int, str)

# String Assignments 


# 1. Write a Python program to get a string from a given string where all occurrences of its first char have been changed to '$', except the first char itself.

Sample String : 'restart'
Expected Result : 'resta$t'

In [23]:
a = '$'
print(f'resta{a}t')

resta$t


# 2. Write a Python program to calculate the length of a string and print a message.
Sample String : 'Hello WOrld!'

In [44]:
h = ("hello world")

len(h), h

(11, 'hello world')

# 3. Write a Python program to get a single string from two given strings, separated by a space and swap the first two characters of each string.
Sample String : 'abc', 'xyz'
Expected Result : 'xyc abz'

In [7]:
message = ('abc', 'xyz')
' '.join(message).replace("abc","xyc").replace("xyz","abz")

'xyc abz'

# 4. Write a Python program to add 'ing' at the end of a given string
Sample String : 'play'
Expected Result : 'playing'

In [26]:
a = "play"
b = "ing" 

print(a+b)

playing


# 5. Write a Python program to find the first appearance of the substrings 'not' and 'poor' in a given string below.
String : 'The lyrics is not that poor! It is Pathetic not poor!'

In [5]:
h = 'the substrings not and poor in a given string below'

h.find('not'), h.find('poor')


(15, 23)

# 6. Write a Python program to remove the nth index character from a nonempty string.

In [30]:
h = "51 545 561 5 1515 1 54 1"
h.split()


['51', '545', '561', '5', '1515', '1', '54', '1']

# 7. Write a Python program to change a given string to a newly string where the first and last chars have been exchanged.
Sample String : 'donkey'
Expected Result : 'yonked'

In [28]:
h = 'donkey'
d= 'y'
y= 'd'
print(f'{d}onke{y}')

yonked


# 8. Write a Python function to get a string made of 4 copies of the last two characters of a specified string (length must be at least 2).
Sample String : 'Python'
Expected Result : 'onononon'

In [35]:
a = 'Python'
a[-2::1]*4

'onononon'

# 9. Write a Python program to get the last character of string

In [38]:
a = 'Python'

a[-1]

'n'

# 10. Write a Python function to reverse a string
Sample String : 'lame'
Expected Result : 'emal'

In [33]:
a = 'lame'

a[-1::-1]

'emal'

# 11. Given two strings, s1 and s2. Write a program to create a new string s3 by appending s2 in the middle of s1.
Given: s1 = 'Tuar', s2 = 'sh'
Expected Output: 'Tushar'

In [45]:
s1 = 'Tuar'
s2 = 'sh' 

s3 = s1[0:2]+s2[0:]+s1[-2:]
s3


'Tushar'

# 12. Given two strings, s1 and s2, write a program to return a new string made of s1 and s2’s first, middle, and last characters.
Given : s1 = 'America', s2 = 'Japan'
Expected Output: 'AJrpan'

In [47]:
s1 = 'America'
s2 = 'Japan' 

s3= s1[0]+s2[0]+s1[3]+s2[2:]
s3

'AJrpan'

# 13. Write a program to count all occurrences of “USA” in a given string ignoring the case.
Given : str1 = "Welcome to USA. usa awesome, isn't it?"
Output : "The USA count is: 2"

In [51]:
str1 = "Welcome to USA. usa awesome, isn't it?"
str2 = "USA"
print(f'The {str2} count is:2')

The USA count is:2


# 14. Write a program to find the last position of a substring “Emma” in a given string. (Hint : use rfind())
Given : "Emma is a data scientist who knows Python. Emma works at google."
Output : 43

In [55]:
str1 = "Emma is a data scientist who knows Python. Emma works at google."
str1.rfind('Emma')

43

# 15. Write a Python program to print the following numbers up to 2 decimal places.
x =  3.1415926
y =  12.9999

In [61]:
x = 22/7
y = 12.9999

round(x,2),round(y,2)

(3.14, 13.0)

# 16. Write a Python program to remove spaces from a given string and convert it into Uppercase.
Given : 'n a r a y an a'
Output : 'NARAYANA'

In [71]:
a = 'n a r a y an a'

b = a.upper().replace(' ','')
b

'NARAYANA'

# If Assignment

# 1. Write a Python Program to check if a number entered by user is odd or even.
I/P : Enter a Number : 5
O/P : 5 is odd number

In [26]:
h = int(input("Enter the number "))

if h%2 == 0:
    print(f'{h} is even number')
else:
    print(f'{h} is odd number')

Enter the number 5446
5446 is even number


# 2. 1. Write a Python Program to check if a number entered by user is prime or not.
I/P : Enter a Number : 7
O/P : 7 is prime number

In [21]:
r = int(input("Enter the number "))
prime = (2, 3, 5, 7, 11, 13, 17, 19, 23, 29, 31, 37, 41, 43, 47, 53, 59, 61, 67, 71, 73, 79, 83, 89, 97)

if r in prime:
    print(f'{r} is prime number')
else:
    print(f'{h} is prime number')

Enter the number 5
5 is prime number


# 3. Create a Python Program to check whether the given string is palindrome or not.
I/P : MOM
O/P: MOM is palindrome

In [27]:
r = input("Enter the name ")

h = r[-1::-1]

if r == h:
    print(f'{r} is pallindrome')
else:
    print(f'{r} is not pallindrome')


Enter the name MOM
MOM is pallindrome


# 4. Create a Program that takes sentence as Input and counts the number of words in it.
I/P: Tushar is teaching at NBS
O/P: 5

In [33]:
a = "Tushar is teaching at NBS"
b = a.split()
len(b)

5

# 5. Given a list of fruits, concatenate them into a single string separated by spaces.
I/P: ["Orange", "Apple", "Grapes"]
O/P: Fruits names are : Orange Apple Grapes

In [1]:
fruits = ["Orange", "Apple", "Grapes"] 

' '.join(fruits)



'Orange Apple Grapes'

# 6. Create a Program that takes a year as input and checks if it is a leap year or not.
I/P : 2020
O/P : 2020 is a leap year

In [2]:
r = int(input("Enter the year "))

if r%4 == 0:
    print(f'{r} is a leap year')
else:
    print(f'{r} is not a leap year')

Enter the year 2056
2056 is a leap year


# 7. Given the email id in the form of a string, check if it is gmail id or not.
I/P : tkakaiya@gmail.com
O/P : It is a gmail id

In [32]:
email = input("Enter gmail id  ")

if "gmail" in email:
    print("it is a gmail id")
else:
    print("it is not a gmail id")

Enter gmail id  tkakaiya@hotmail.com
it is not a gmail id


# 8. Given the email id in the forma of a string, find which domain id it is.
I/P : tush_ar25@yahoo.com
O/P : It is yahoo domain email ID.

In [1]:
email = input("Enter gmail id  ")

a = email[-9:-4:1]
a

Enter gmail id  harsh@gmail.com


'gmail'

# 9. Write a Program that takes a string from user and print that string only if it contains more than 5 characters.
I/P : Tushar
O/P : Hi Tushar
 
I/P : RAAM
O/P : Not a 5 character string!

In [8]:
h = input("Enter your name ")


if len(h)>=5:
    print("There is the 5 character sting!")
else:
    print("There is Not a 5 character string!")

Enter your name hars
There is Not a 5 character string!


# 10. Write a Program to build a calculator based on the operators and the operands given
Do it for +, -, *, /, //, %
I/P :
Enter the value of first operand: 8
Enter the value of second operand: 9
Enter the Operator : +
Answer of addition of 8 and 9 is : 17
 
Enter the value of first operand: 8
Enter the value of second operand: 9
Enter the Operator : *
Answer of multiplication of 8 and 9 is : 72

In [10]:
h1 = int(input("Enter the first value "))
h2 = int(input("Enter the second value "))
h3 = input("Enter the Operator ")

if h3 == "+":
    print(h1+h2)
elif h3 == "*":
    print(h1*h2)
elif h3 == "-":
    print(h1-h2)
elif h3 == "/":
    print(h1/h2)
elif h3 == "//":
    print(h1//h2)
elif h3 == "%":
    print(h1%h3)
else:
    print("sorry you can do only +,-,*,/,//,%")

Enter the first value 5
Enter the second value 8
Enter the Operator +
13


# 11. Implement a Program that takes a sentence and a word as Input and checks if the word is present in the sentence (Hint : Use split and membership test operators)
I/P :
Enter the statement: Tushar is teaching at NBS
Enter the word to search : working
Result: Match not found!

In [12]:
a = input("Enter the sentence ")
b = " Tushar is teaching at NBS"

if a in b:
    print("Match is found")
else:
    print("Match is not found")

Enter the sentence Tushar
Match is found


# 12. Write a Program which takes a number from the user and find whether that number is divisible by 7 and multiples of 5 or not.
I/P : Enter the number : 1505
O/P : Yes, its divisible by 7 and multiples of 5
 
I/P : Enter the number : 1715
O/P : Yes, its divisible by 7 and multiples of 5
 
I/P : Enter the number : 1700
O/P : No, its not divisible by 7 and not multiples of 5

In [10]:
r = int(input("Enter the number "))
h = r%7
a = r%5

if h == a:
    print("Yes, its divisible by 7 and multiples of 5")
else:
    print("NO, its not divisible by 7 and not multiples of 5")

Enter the number 1700
NO, its not divisible by 7 and not multiples of 5


# 13. Write a Python program to check whether an alphabet is a vowel or consonant.
I/P :
Input a letter of the alphabet: k
O/P:
k is a consonant.
 
I/P :
Input a letter of the alphabet: i
O/P:
i is a vowel.

In [14]:
h = input("Enter an alphabet ")

vowel = "a,e,i,o,u"

if h in vowel:
    print(f"{h} is vowel")
else:
    print(f"{h} is consonant")

Enter an alphabet k
k is consonant


# 14. Write a Python program to sum two integers. However, if the sum is between 15 and 20 it will print 20.

In [8]:
h = int(input("Enter first an number "))
b = int(input("Enter second an number "))
a = h+b

if a >=15 and a <=20:
    print("20")
else:
    print("Sorry")

Enter first an number 8
Enter second an number 8
20


# 15. Write a Python program to check if a triangle is equilateral, isosceles or scalene.
Note :
An equilateral triangle is a triangle in which all three sides are equal.
A scalene triangle is a triangle that has three unequal sides.
An isosceles triangle is a triangle with (at least) two equal sides.
I/P:
Input lengths of the triangle sides:
x: 6
y: 8
z: 12
O/P:
Scalene triangle

In [17]:
x = int(input("Enter first side of tringle "))
y = int(input("Enter second side of traingle "))
z = int(input("Enter second side of traingle "))

if x == y== z:
    print("it's an equilateral tringle")
elif x==y or x==z or y==x or y==z or z==x or z==y:
    print("it's a two equl(isosceles) tringle")
else:
    print("it's a scalene tringle")

Enter first side of tringle 5
Enter second side of traingle 6
Enter second side of traingle 3
it's a scalene tringle


# 16. Write a Python program to display the astrological sign for a given date of birth.
I/P:
Input birthday: 15
Input month of birth (e.g. march, july etc): may
O/P:
Your Astrological sign is : Taurus

In [3]:
day = int(input("Enter you birthday "))
month = input("Enter you birth month ")

if (21<= day <= 31 and month == "March") or (1 <= day <= 19 and month == "April"):
    print("Your Astrological sign is: Aries")
elif (20<= day <= 30 and month == "April") or (1 <= day <= 20 and month == "May"):
    print("Your Astrological sign is: Taurus")
elif (21<= day <= 31 and month == "May") or (1 <= day <= 21 and month == "June"):
    print("Your Astrological sign is: Gemini")
elif (22<= day <= 30 and month == "June") or (1 <= day <= 22 and month == "July"):
    print("Your Astrological sign is: Cancer")
elif (23<= day <= 31 and month == "July") or (1 <= day <= 22 and month == "August"):
    print("Your Astrological sign is: Leo")
elif (23<= day <= 31 and month == "August") or (1 <= day <= 23 and month == "September"):
    print("Your Astrological sign is: Virgo")
elif (23<= day <= 30 and month == "September") or (1 <= day <= 23 and month == "October"):
    print("Your Astrological sign is: Libra")
elif (24<= day <= 31 and month == "October") or (1 <= day <= 21 and month == "November"):
    print("Your Astrological sign is: Scorpio")
elif (22<= day <= 30 and month == "November") or (1 <= day <= 21 and month == "December"):
    print("Your Astrological sign is: Sagittarius")
elif (22<= day <= 31 and month == "December") or (1 <= day <= 19 and month == "January"):
    print("Your Astrological sign is: Capricorn")
elif (20 <= day <= 31 and month == "January") or (1 <= day <= 18 and month == "February"):
    print("Your Astrological sign is: Aquarius")
elif (19<= day <= 29 and month == "February") or (1 <= day <= 20 and month == "March"):
    print("Your Astrological sign is: Pisces")

Enter you birthday 12
Enter you birth month August
Your Astrological sign is: Leo


# 17. Write a Python program to find the median of three values.
I/P:
Input first number: 15
Input second number: 26
Input third number: 29
 
O/P:
The median is 26.0

In [11]:
a = int(input("Enter the first value "))
b = int(input("Enter the second value "))
c = int(input("Enter the first value "))

if b < a < c:
    print(f"The median {a}")
elif a < b < c:
    print(f"The median {b}")
else:
    print(f"The median {c}")
    

Enter the first value 15
Enter the second value 26
Enter the first value 29
The median 26


# 18. A company decided to give bonus of 5% to employee if his/her year of service is more than 5 years.
Ask user for their salary and year of service and print the net bonus amount.

I/P:
Enter Salary : 10000
Enter Years of Service : 6
O/P:
Bonus is : 500
 
I/P:
Enter Salary : 10000
Enter Years of Service : 3
O/P:
Sorry, you are not eligible for bonus!

In [18]:
salary =  int(input("Enter your salary amount "))
years = int(input("Enter the years of service "))

if years > 5:
    print("Bonus",salary*5/100)
else:
    print("you are not eligible for bounus")

Enter your salary amount 10000
Enter the years of service 6
Bonus 500.0


# 19. NBS has following rules for grading system:

a. Below 25 - F,
b. 25 to 45 - E,
c. 45 to 50 - D,
d. 50 to 60 - C,
e. 60 to 80 - B,
f. Above 80 - A,
Ask user to enter marks and print the corresponding grade.

In [5]:
n = int(input("Enter a number "))

if 0<= n <= 25:
    print("f")
elif 25<= n <= 45:
    print("E")
elif 45<= n <= 50:
    print("D")
elif 50 <= n <= 60:
    print("C")
elif 60<= n <= 80:
    print("B")
elif 80<= n:
    print("A")

Enter a number 85
A


# 21. A student will not be allowed to sit in exam if his/her attendence is less than 75%.
Take following input from user:
Number of classes held
Number of classes attended.
And print
percentage of class attended
And answer Is student is allowed to sit in exam or not.

In [9]:
ch = int(input("Enter the number of clasees held "))
ca = int(input("Enter the number of clasees attened "))

a = ca/ch*100

print(f'{a}%')

if a >= 75:
    print("You can seat in the exam")
else:
    print("You can not seat in the exam")

Enter the number of clasees held 100
Enter the number of clasees attened 76
76.0%
You can seat in the exam


# 22. Take values of length and breadth of a rectangle from user and check if it is square or not.
 

In [20]:
length = int(input("Enter the lenght "))
breadth = int(input("Enter the breadth "))

h = length * breadth

print(f"Area of rectangle is {h}")

if length == breadth:
    print("It is square")
else:
    print("It is not square")

Enter the lenght 45
Enter the breadth 90
Area of rectangle is 4050
It is not square


# 23. Take two int values from user and print greatest among them.
 

In [23]:
n = int(input("Enter a number "))
h = int(input("Enter a number "))

if n > h:
    print(n)
else:
    print(h)

Enter a number 45
Enter a number 66
66


# 24. Take input of age of 3 people by user and determine oldest and youngest among them.

In [32]:
n = int(input("Enter the fist person's age "))
h = int(input("Enter the second person's age "))
b = int(input("Enter the third person's age "))

g = [n,h,b]

g.sort()
g[2]

Enter the fist person's age 15
Enter the second person's age 145
Enter the third person's age 45


145

# 25. Write a Program to print "Hello" if a number entered by user is divisible by 7 otherwise print "Bye"


In [36]:
n = int(input("Enter a number "))

if n%7 == 0:
    print("Hello")
else:
    print("Bye")

Enter a number 85
Bye


# For Loop Assignment

# 1. Print first 10 numbers using a for loop

In [10]:
h = list(range(0,11,1))

for i in h:
    print(i)

0
1
2
3
4
5
6
7
8
9
10


# 2. Calculate the square of each number of list of 1-100 numbers

In [3]:
r = int(input("Enter the number "))
h = list(range(0,r+1,1))
a = []

for i in h:
    a.append(i**2)
    
print(a)

Enter the number 100
[0, 1, 4, 9, 16, 25, 36, 49, 64, 81, 100, 121, 144, 169, 196, 225, 256, 289, 324, 361, 400, 441, 484, 529, 576, 625, 676, 729, 784, 841, 900, 961, 1024, 1089, 1156, 1225, 1296, 1369, 1444, 1521, 1600, 1681, 1764, 1849, 1936, 2025, 2116, 2209, 2304, 2401, 2500, 2601, 2704, 2809, 2916, 3025, 3136, 3249, 3364, 3481, 3600, 3721, 3844, 3969, 4096, 4225, 4356, 4489, 4624, 4761, 4900, 5041, 5184, 5329, 5476, 5625, 5776, 5929, 6084, 6241, 6400, 6561, 6724, 6889, 7056, 7225, 7396, 7569, 7744, 7921, 8100, 8281, 8464, 8649, 8836, 9025, 9216, 9409, 9604, 9801, 10000]


# 3. Calculate the sum of all even numbers from 1-100

In [36]:
h = list(range(0,101,2))
a = 0
for i in h:
    a = i+ a

print(a)

2550


# 4. Write a Program to print the table of the number entered by user.

In [18]:
h = int(input("Enter the number will make table of that table "))

range1 = list(range(1,11,1))
f = []

for i in range1:
    f.append(h*i)
    
print(f)

Enter the number will make table of that table 5
[5, 10, 15, 20, 25, 30, 35, 40, 45, 50]


In [16]:
tab = int(input("Enter the number will make table of that table "))
ans = 0

for i in range(1,11):
    ans = tab *i
    print(f"{tab} * {i} = {ans}")

Enter the number will make table of that table 5
5 * 1 = 5
5 * 2 = 10
5 * 3 = 15
5 * 4 = 20
5 * 5 = 25
5 * 6 = 30
5 * 7 = 35
5 * 8 = 40
5 * 9 = 45
5 * 10 = 50


# 5. Write a Program to find the factorial of the number enetered by user.
(Hint: 5 factorial is 5*4*3*2*1)
I/P : Enter the number : 5
O/P: Factorial of 5 is 120

In [68]:
h = int(input("Enter the number "))
a = list(range(h,0,-1))
s = 1 

for i in a:
    s = (s*i)
    
print(s)

Enter the number 5
120


# 6. Write a Program to check if the number is prime or not (Hint: Prime number is divisble by 1 and the number itself!)

In [30]:
num = int(input("Enter the number "))
range1 = range(1,num)
r = 0

for i in range1:
    if num % i == 0:
        r = i
        

if r == 0:
    print(f"{num} is prime number")
else:
    print(f"{num} is not prime number")
        

Enter the number 5
5 is not prime number


# 7. Write a Program to print sum of all odd number between 12 and 37 (12 and 37 included)

In [73]:
a =list(range(13,38,2))

b = [12]

for i in a:
    b.append(i)

sum(b)

337

# 8. Write a Program to find whether the number entered by user is perfect number or not (Hint: Perfect number is the number where the divisors sum is the number itself!)
For example, 6 is perfect number because the divisors of 6 are : 1,2,3 and 1+2+3 is 6

In [66]:
h = int(input("Enter the number "))
a = list(range(1,h,1))
r = []

for i in a:
    if h % i == 0:
        r.append(i)

g = 0
for j in r:
    g = g+j

if g ==h:
    print(f"Yes, {h} is percet number")
else:
    print(f"No, {h} is percet number")

Enter the number 6
Yes, 6 is percet number


# 9. Write a Program to find the sum of following (accepted values a,r and n from user)
a + ar + ar2 + ar3 + .... + arn

In [2]:
a = int(input("Enter the number a "))
r = int(input("Enter the number r "))
n = int(input("Enter the number n "))

sum2 = 0

for i in range(0,n+1,1):
    sum2 = sum2 +a*r**i
    
print(sum2)

Enter the number a 5
Enter the number r 2
Enter the number n 4
155


# 10. Write a Program in python to append suffix in each word of the string.
I/P: "Play Work Sleep Eat"
O/P: "Playing Working Sleeping Eating"

In [32]:
work = ("Play Work Sleep Eat")
work1 = work.split(" ")
ans = []

for i in work1:
    ans.append(i+"ing")

ans
    
    

['Playing', 'Working', 'Sleeping', 'Eating']

# 11. Display Fibonacci series up to 10 terms
Example: 1,2,3,5,8,13,21,34

In [1]:
h = int(input("Enter the number "))
rang1 = list(range(0,h-2,1))

final1 =[1,2]

for i in rang1:
    final1.append(final1[-1]+final1[-2])

print(final1)
    


Enter the number 10
[1, 2, 3, 5, 8, 13, 21, 34, 55, 89]


# 12. Calculate the cube of all numbers from 1 to a given number

In [4]:
h = int(input("Enter the range "))
a = list(range(1,h+1,1))
j = []

for i in a:
    j.append(i**3)
    
print(j)
    

Enter the range 5
[1, 8, 27, 64, 125]


# 13. Display numbers from -10 to -1 using for loop

In [34]:
for i in range(-10,0,1):
         print(i) 

-10
-9
-8
-7
-6
-5
-4
-3
-2
-1


# 14. Count the total number of digits in a number

In [4]:
num = 9157533170
count=0
for i in str(num):
    count=count+1
count

10

# 15. Print "Done" after successful execution of for loop
Sample O/P:
0
1
2
3
4
Done!

In [21]:
h = int(input("Enter the range "))
a = []
for i in range(0,h+1,1):
    a.append(i)
    print(i)

print("Done!")

Enter the range 5
0
1
2
3
4
5
Done!


# 17. Write a Program to check if a given string is a pangram. (contains all letters of the alphabet)

In [9]:
abc = input("Enter any alphabet ")
all = "qwertyuiopasdfghjklzxcvbnm"
ans = []

for i in all:
    if i in abc:
        ans.append("y")
    else:
        ans.append("n")
        
if ans == "y":
    print("String isn't Pangram")
else:
    print("String is Pangram")

Enter any alphabet qwertyuiopasdfghjklzxcvbnm
String is Pangram


# 18. Given a list of names, print all names starting with the letter 'A'

In [21]:
nlist=["Darshan","Harsh","Chirag","Jayesh","Amisha","Prem","Ayush","Preet","Arpit"]
name = []


for i in nlist:
    if i[0:1:1] == "A":
        name.append(i)
name

['Amisha', 'Ayush', 'Arpit']

# 19. Create a loop that prints all prime numbers between 1 and 50.

In [32]:
range1 = list(range(1,51,1))
ind_range = []

for i in range1:
    ind_range = list(range(1,i))

ind_range
        

[1,
 2,
 3,
 4,
 5,
 6,
 7,
 8,
 9,
 10,
 11,
 12,
 13,
 14,
 15,
 16,
 17,
 18,
 19,
 20,
 21,
 22,
 23,
 24,
 25,
 26,
 27,
 28,
 29,
 30,
 31,
 32,
 33,
 34,
 35,
 36,
 37,
 38,
 39,
 40,
 41,
 42,
 43,
 44,
 45,
 46,
 47,
 48,
 49]

# 20. Given a list of words, count the number of words with more than 5 characters.

In [46]:
nlist=["Darshan","Harsh","Chirag","Jayesh","Amisha","Prem","Ayush","Preet","Arpit"]
ans = []

for i in nlist:
    if len(i) == 5:
        ans.append(i)
        
ans

['Harsh', 'Ayush', 'Preet', 'Arpit']

# 21. Calculate the sum of all digits given in a string format
I/P : s= "12345"
O/P : 15

In [41]:
h = "12345"
a = 0

for i in h:
    a = int(i)+a 
    
a
    

15

# Dictionary Assignment 

# Q-1. Write a Python script to sort (ascending and descending) a dictionary by value.

In [19]:
d={0: 10, 1: 20,2:9,3:23,4:8}
d.values()

dict_values([10, 20, 9, 23, 8])

# Q-2. Write a Python script to add a key to a dictionary.
Sample Dictionary : {0: 10, 1: 20}
Expected Result : {0: 10, 1: 20, 2: 30}


In [9]:
list1 = {0: 10, 1: 20}

list1[2] = 30

list1

{0: 10, 1: 20, 2: 30}

# Q-3. Write a Python script to concatenate the following dictionaries to create a new one.
 
Sample Dictionary :
dic1={1:10, 2:20}
dic2={3:30, 4:40}
dic3={5:50,6:60}
Expected Result : {1: 10, 2: 20, 3: 30, 4: 40, 5: 50, 6: 60}

In [109]:
dic1={1:10, 2:20} 
dic2={3:30, 4:40} 
dic3={5:50,6:60}
ans ={}

for i in [dic1,dic2,dic3]:
    ans.update(i)
ans
    

{1: 10, 2: 20, 3: 30, 4: 40, 5: 50, 6: 60}

# Q-4. Write a Python script to check whether a given key already exists in a dictionary.If exists then print the value corresponding to that key else print "key not found" error message.
(Use: Try..Except block)

In [27]:
ans = {1: 10, 2: 20, 3: 30, 4: 40, 5: 50, 6: 60}
ans.get(7, "key not found")

'key not found'

# Q-5. Write a Python script to generate and print a dictionary that contains a number (between 1 and n) in the form (x, x^3). Take n from user
Sample Dictionary ( n = 5) :
Expected Output : {1: 1, 2: 8, 3: 27, 4: 64, 5: 125}
 

In [175]:
num = int(input("Enter the number "))
num1 =  []
ans = []


for i in range(1,num+1,1):
        num1.append(i)
        
for e in num1:
    ans.append(e**3)

ans1 = dict(zip(num1,ans))
ans1

Enter the number 5


{1: 1, 2: 8, 3: 27, 4: 64, 5: 125}

# Q-6. Write a Python script to print a dictionary where the keys are numbers between 1 and 15 (both included) and the values are the square of the keys.
Sample Dictionary
{1: 1, 2: 4, 3: 9, 4: 16, 5: 25, 6: 36, 7: 49, 8: 64, 9: 81, 10: 100, 11: 121, 12: 144, 13: 169, 14: 196, 15: 225}

In [178]:
num = int(input("Enter the number "))
num1 =  []
ans = []


for i in range(1,num+1,1):
        num1.append(i)
        
for e in num1:
    ans.append(e**2)

ans1 = dict(zip(num1,ans))
ans1

Enter the number 10


{1: 1, 2: 4, 3: 9, 4: 16, 5: 25, 6: 36, 7: 49, 8: 64, 9: 81, 10: 100}

# Q-7. Convert two lists into a dictionary
Sample Input:
keys = ['Ten', 'Twenty', 'Thirty']
values = [10, 20, 30]
Expected output:
{'Ten': 10, 'Twenty': 20, 'Thirty': 30}

In [180]:
keys = ['Ten', 'Twenty', 'Thirty']
values = [10, 20, 30]

ans = dict(zip(keys,values))
ans

{'Ten': 10, 'Twenty': 20, 'Thirty': 30}

# Q-8. Merge two Python dictionaries into one
Sample Input:
dict1 = {'Ten': 10, 'Twenty': 20, 'Thirty': 30}
dict2 = {'Thirty': 30, 'Fourty': 40, 'Fifty': 50}
 
Expected output:
{'Ten': 10, 'Twenty': 20, 'Thirty': 30, 'Fourty': 40, 'Fifty': 50}

In [186]:
dict1 = {'Ten': 10, 'Twenty': 20, 'Thirty': 30}
dict2 = {'Thirty': 30, 'Fourty': 40, 'Fifty': 50}

dict1["Fourty"] = 40
dict1['Fifty'] = 50
dict1


{'Ten': 10, 'Twenty': 20, 'Thirty': 30, 'Fourty': 40, 'Fifty': 50}

# Q-9. Delete a list of keys from a dictionary
Given:
 
sample_dict = {
"name": "Kelly",
"age": 25,
"salary": 8000,
"city": "New york"
}
 
# Keys to remove
keys = ["name", "salary"]
Expected output:
 
{'city': 'New york', 'age': 25}
 

In [191]:
sample_dict = {"name": "Kelly", "age": 25, "salary": 8000, "city": "New york"}

del sample_dict["name"]
del sample_dict["salary"]
sample_dict

{'age': 25, 'city': 'New york'}

# Q-10. Rename key of a dictionary
Write a program to rename a key city to a location in the following dictionary.
 
Given:
 
sample_dict = {
"name": "Kelly",
"age":25,
"salary": 8000,
"city": "New york"
}
Expected output:
 
{'name': 'Kelly', 'age': 25, 'salary': 8000, 'location': 'New york'}

In [212]:
sample_dict = {"name": "Kelly", "age": 25, "salary": 8000, "city": "New york"}

del sample_dict["city"]
sample_dict["location"]= "New york"
sample_dict 

{'name': 'Kelly', 'age': 25, 'salary': 8000, 'location': 'New york'}

# Q-11. Get the key of a minimum value from the following dictionary
sample_dict = {
'Physics': 82,
'Math': 65,
'history': 75
}
Expected output:
 
Math

In [19]:
sample_dict = {'Physics': 82,'Math': 65,'history': 75}

list2 = list(sample_dict.keys())
list1 =  list(sample_dict.values())

min1 = list1.index(min(list1))
list2[min1]

'Math'

In [8]:
sample_dict = {'Physics': 82,'Math': 65,'history': 75}
min(sample_dict)

'Math'

# Q-12. Write a Python program to change Brad’s salary to 8500 in the following dictionary.
 
sample_dict = {
'emp1': {'name': 'Jhon', 'salary': 7500},
'emp2': {'name': 'Emma', 'salary': 8000},
'emp3': {'name': 'Brad', 'salary': 500}
}
Expected output:
 
{
'emp1': {'name': 'Jhon', 'salary': 7500},
'emp2': {'name': 'Emma', 'salary': 8000},
'emp3': {'name': 'Brad', 'salary': 8500}
}
 

In [257]:
sample_dict = { 'emp1': {'name': 'Jhon', 'salary': 7500}, 
               'emp2': {'name': 'Emma', 'salary': 8000}, 
               'emp3': {'name': 'Brad', 'salary': 500} }

sample_dict["emp3"]["salary"]= 8500
sample_dict

{'emp1': {'name': 'Jhon', 'salary': 7500},
 'emp2': {'name': 'Emma', 'salary': 8000},
 'emp3': {'name': 'Brad', 'salary': 8500}}

# Q-13. Print the value of key ‘history’ from the below dict
 
Given:
sampleDict = {
"class": {
"student": {
"name": "Mike",
"marks": {
"physics": 70,
"history": 80
}
}
}
}
 
Expected output:
 
80

In [265]:
sampleDict = { "class": { "student": { "name": "Mike", "marks": { "physics": 70, "history": 80 } } } }

sampleDict["class"]["student"]["marks"]["history"]

80

# Q-14. Write a Python program to combine two dictionary by adding values for common keys.
d1 = {'a': 100, 'b': 200, 'c':300}
d2 = {'a': 300, 'b': 200, 'd':400}
Sample output: Counter({'a': 400, 'b': 400, 'd': 400, 'c': 300})

In [355]:
d1 = {'a': 100, 'b': 200, 'c':300}
d2 = {'a': 300, 'b': 200, 'd':400}


d2["a"] = 400
d2["b"] = 400
d2["c"] = 300
d1["d"] = 400

ans = dict(zip(d1,d2.values()))
ans

{'a': 400, 'b': 400, 'c': 400, 'd': 300}

In [20]:
d1 = {'a': 100, 'b': 200, 'c':300}
d2 = {'a': 300, 'b': 200, 'd':400}
d11=list(d1.items())
d22=list(d2.items())

for i in range(len(d11)):
    if d11[i][0] ==d22[i][0]:
        d1.update({d11[i][0]:d11[i][1]+d22[i][1]})
    else:
        d1.update({d11[i][0]:d11[i][1]})
        d1.update({d22[i][0]:d22[i][1]})
d1
       

{'a': 400, 'b': 400, 'c': 300, 'd': 400}

# Q-15. Write a Python program to print all distinct values in a dictionary.
Sample Data : [{"V":"S001"}, {"V": "S002"}, {"VI": "S001"}, {"VI": "S005"}, {"VII":"S005"}, {"V":"S009"},{"VIII":"S007"}]
Expected Output : Unique Values: {'S005', 'S002', 'S007', 'S001', 'S009'}

In [387]:
SampleData = [{"V":"S001"}, {"V": "S002"}, {"VI": "S001"}, {"VI": "S005"}, {"VII":"S005"}, {"V":"S009"},{"VIII":"S007"}]

a = {SampleData[0]["V"],SampleData[1]["V"],SampleData[2]["VI"],SampleData[3]["VI"],SampleData[4]["VII"],SampleData[5]["V"],SampleData[6]["VIII"]}
a

{'S001', 'S002', 'S005', 'S007', 'S009'}

# Q-16. Write a Python program to combine values in a list of dictionaries.
Sample data: [{'item': 'item1', 'amount': 400}, {'item': 'item2', 'amount': 300}, {'item': 'item1', 'amount': 750}]
Expected Output: Counter({'item1': 1150, 'item2': 300})

In [1]:
d=[{'item': 'item1', 'amount': 400}, {'item': 'item2', 'amount': 300}, {'item': 'item1', 'amount': 750}]
dd=[]
for i in d:
    dd.append(list(i.values()))
ddd=[]
if dd[0][0]==dd[1][0]:
    ddd.append([dd[0][0],dd[0][1]+dd[1][1]])
    ddd.append([dd[2][0],dd[2][1]])
elif dd[1][0]==dd[2][0]:
    ddd.append([dd[1][0],dd[1][1]+dd[2][1]])
    ddd.append([dd[0][0],dd[0][1]])
elif dd[0][0]==dd[2][0]:
    ddd.append([dd[0][0],dd[0][1]+dd[2][1]])
    ddd.append([dd[1][0],dd[1][1]])
ddd=dict(ddd)
ddd

{'item1': 1150, 'item2': 300}

In [7]:
Sampledata= [{'item': 'item1', 'amount': 400}, {'item': 'item2', 'amount': 300}, {'item': 'item1', 'amount': 750}]

Sampledata[0]["amount"]= 1150
del Sampledata[2]
Sampledata

[{'item': 'item1', 'amount': 1150}, {'item': 'item2', 'amount': 300}]

# Functions Assignment

# Q -1.Write a program that asks the user for two numbers. Then ask them if they would like to add, subtract, divide, or multiply these numbers. Perform the chosen operation on the values, showing the operation being performed. Write four functions, one for each mathematical operation.
Example: add(), subtract(), Multiply(), and Divide() functions should be written to perform calculations.

In [2]:
def addtion(int1,int2):
    return int1 +int2 

addtion(4,5)

9

In [4]:
def subtract(int1 , int2):
    return int1 - int2

subtract(10,6)

4

In [8]:
def divide(int1,int2):
    return int1/int2

divide(10,5)

2.0

In [9]:
def multiply(int1,int2):
    return int1 * int2

multiply(5,2)

10

# Q-2. Write a function called favorite_book() that accepts two parameter, title and author. The function should print a message, such as "Ansi C by E Balagurusamy". Call the function, make sure to include a book title and author as an argument in the function call.

In [18]:
def favorite_book(str1,str2):
    return str1 + " "+ "by"+" "+ str2

favorite_book("Ansi C","E Balagurusamy")

'Ansi C by E Balagurusamy'

# Q-3. Write a function called is_leap() that accepts one parameter i.e. year. The function should check whether the year is a leap year or not.

In [35]:
def is_leap(int1):
    if int1 % 4 == 0:
       return f"{int1} is yes"
    else:
        return f"{int1} is No"
        
is_leap(2024)

'2024 is yes'

# Q-4. Write a Python function to find the maximum of three number.


In [50]:
def max(int1,int2,int3):
    if int1 >int2 and int1 > int3:
        return int1
    elif int2 > int1 and int2 > int3:
        return int2
    else:
        return int3
    


max(4888,4,8)

4888

# Q-5. Write a Python function to multiply all the numbers in a list.
Sample List : (8, 2, 3, -1, 7)
Expected Output : -336

In [1]:
def mul(*int):
    ans = 1
    for i in int:
        ans = i * ans
    return ans

mul(2,3,54,54,5)

87480

# Q-6. Write a Python program to reverse a string.
Sample String : "1234abcd"
Expected Output : "dcba4321"

In [21]:
def rev(str):
    list1= list(str)
    ans = list1[-1::-1]
    return ''.join(ans)

rev("1234abcd")

'dcba4321'

# Q-7. Write a Python function to calculate the factorial of a number (a non-negative integer). The function accepts the number as an argument.

In [11]:
def fac(int1):
    list1 = 1
    for i in list(range(int1,0,-1)):
        list1 = list1*i
    return list1

fac(5)

120

# Q-8. Write a Python function that accepts a string and counts the number of upper and lower case letters.
Sample String : 'The quick Brow Fox'
Expected Output :
No. of Upper case characters : 3
No. of Lower case Characters : 12

In [1]:
def uplo(st):
    up="QWERTYUIOPASDFGHJKLZXCVBNM"
    lo="qwertyuiopasdfghjklzxcvbnm"
    nup=0
    nlo=0
    for i in st:
        if i in up:
            nup+=1
        if i in lo:
            nlo+=1
    print(f"No. of Upper case characters : {nup}")
    print(f"No. of Lower case Characters : {nlo}")

uplo('Harsh')

No. of Upper case characters : 1
No. of Lower case Characters : 4


# Q-9. Write a Python function that takes a list and returns a new list with distinct elements from the first list.
 

In [28]:
def dis(*str):
    str_set = {*str}
    return str_set

dis("Harsh","Harsh")

{'Harsh'}

# Q-10. Write a Python program to print the even numbers from a given list.

In [54]:
def even(*int):
    ans = []
    for i in int:
        if i%2 == 0:
            ans.append(i)
            return ans

even(1,5,3,8,6)


[8]

# Q-11. Write a Python function that checks whether a passed string is a palindrome or not.
Note: A palindrome is a word, phrase, or sequence that reads the same backward as forward, e.g., madam or nurses run.

In [55]:
def pali(str):
    if str[-1::-1]==str:
        return "yes"
    else:
        return "no"

pali("MOM")

'yes'

# Q-12. Write a Python function to check whether a string is a pangram or not.
Note : Pangrams are words or sentences containing every letter of the alphabet at least once.
For example : "The quick brown fox jumps over the lazy dog"

In [2]:
def all(typ):
    h = "q w e r t y u i o p a s d f g h j k l z x c v b n m"
    j = h.split()
    r =typ.replace(' ','')
    e = r.replace('',' ').split()
    
    for i in h:
        if i in e:
            return "Yes"
        else:
            return "No"

all("The quick brown fox jumps over the lazy dog")

'Yes'

# Q-13. Write a Python program that accepts a hyphen-separated sequence of words as input and prints the words in a hyphen-separated sequence after sorting them alphabetically.¶
Sample Items : green-red-yellow-black-white Expected Result : black-green-red-white-yellow

In [2]:
def sortbyhyphen(st):
    sp=st.split("-")
    sp=tuple(sorted(sp))
    sp="-".join(sp)
    print(sp)

sortbyhyphen("green-red-yellow-black-white")

black-green-red-white-yellow


# Q-14. Write a Python function to create and print a list where the values are the squares of numbers between 1 and 30 (both included).

In [2]:
def seq(int):
    ran= list(range(1,int+1,1))
    ans = []
    
    for i in ran:
        ans.append(i**3)
    return ans
seq(5)

[1, 8, 27, 64, 125]

# Q-15. Write a program to create a function show_employee() using the following conditions.¶
It should accept the employee’s name and salary and display both. If the salary is missing in the function call then assign default value 9000 to salary

In [ ]:
def show_employee(str,int=9000):
    return print(str,int)

show_employee("harsh",)

# Q-16. Write a program to swap two numbers.
Before swap : 17 24 After swap : 24 17 Solution given below: (Just for your reference) def SwapTwoNumbers(a,b): print("Before Swap: ",a,b) a = a+b b = a-b a = a-b return a,b a,b = SwapTwoNumbers(17,24) print("After Swap: ",a,b)

In [5]:
a = 1 + 2
b = a - 2
a = a - b

a,b

(2, 1)

In [6]:
def swap(int1,int2):
    int1 = int1 + int2
    int2 = int1 - int2
    int1 = int1 - int2
    return int1,int2

swap(14,66)

(66, 14)